tools/reviewer/remote_server/RemoteServer.java

```diff
@@ -0,0 +1,208 @@
+      return parseGoogleResponse(response);
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+      httpExchange.getResponseHeaders().add("Access-Control-Allow-Origin", "*");
+      httpExchange
+          .getResponseHeaders()
+          .add("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+      if ("post".equalsIgnoreCase(httpExchange.getRequestMethod())) {
+        logger.atInfo().log("Handling initial auth request");
+
+        InitialAuthRequest.Builder req = InitialAuthRequest.newBuilder();
+        JsonFormat.parser().merge(getPostParamsString(httpExchange), req);
+        AuthResponse response = null;
+
+        try {
+          response = codeExchangeRequest(req.build());
+        } catch (Exception e) {
+          e.printStackTrace();
+          httpExchange.sendResponseHeaders(HTTP_STATUS_CODE_UNAUTHORIZED, -1);
+        }
+
+        sendProto(httpExchange, response);
+      }
+    }
+  }
+
+  static class RefreshTokenHandler implements HttpHandler {
+    private String clientId;
+    private String clientSecret;
+
+    RefreshTokenHandler(String clientId, String clientSecret) {
+      this.clientId = clientId;
+      this.clientSecret = clientSecret;
+    }
+
+    private String buildRefreshParams(String refreshToken) {
+      return String.format(
+          "client_id=%s&client_secret=%s&refresh_token=%s&grant_type=refresh_token",
+          this.clientId, this.clientSecret, refreshToken);
+    }
+
+    private AuthResponse tokenRefreshRequest(RefreshTokenRequest request) throws Exception {
+      URL url = new URL(RemoteServer.TOKEN_URL);
+      String response = httpPost(url, buildRefreshParams(request.getRefreshToken()));
+      return parseGoogleResponse(response);
+    }
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+      httpExchange.getResponseHeaders().add("Access-Control-Allow-Origin", "*");
+      httpExchange
+          .getResponseHeaders()
+          .add("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+      if ("post".equalsIgnoreCase(httpExchange.getRequestMethod())) {
+        logger.atInfo().log("Handling refresh token request");
+
+        RefreshTokenRequest.Builder req = RefreshTokenRequest.newBuilder();
+        JsonFormat.parser().merge(getPostParamsString(httpExchange), req);
+        AuthResponse response = null;
+        try {
+          response = tokenRefreshRequest(req.build());
+        } catch (Exception e) {
+          e.printStackTrace();
+          httpExchange.sendResponseHeaders(HTTP_STATUS_CODE_UNAUTHORIZED, -1);
+        }
+
+        sendProto(httpExchange, response);
+      }
+    }
+  }
+
+  private RemoteServer(String clientId, String clientSecret) throws IOException {
+    httpServer =
+        HttpServer.create(new InetSocketAddress(Integer.parseInt(System.getenv("PORT"))), 0);
```

Comment:
Let's hardcode it or use a flag. I don't like the use of env-vars. They make code less reproducible.